### PR TITLE
refactor Endpoint http methods to _private, return request object

### PR DESCRIPTION
**Pull request by pocin**
**This is exactly the same as #32 but rebased + all commits squashed into a big one**, #32 was impossible to merge

--- 
To avoid collisions with the endpoint names and avoid using super().delete(...)
because that is just confusing.

refactor authentication headers handling to Endpoint class
To avoid brutal code duplication. This still allows passing custom headers, to
the child endpoint methods if required.
bug: fix table data-preview made unauthorized requests
remove Makefile
basic endpoint http methods return response object, not json
chore: remove unused import, add missing whitespace
convert line endings to unix to ease merging to master

### CODE REVIEW

**Comment by odinuv** [comment]
I still think that something like this
```
class Endpoint:
     def _get_response(self, url, params, **kwargs):
            r = requests.get(url, params, **kwargs)
            try:
                r.raise_for_status()
            except requests.HTTPError:
                # Handle different error codes
                raise
            else:
                return r

    def _get(self, *args, **kwargs):
        return self._get_response(*args,**kwargs).json()
```

is far better than having `.json` in every request - i'd like to hear what others think - @Ogaday @pivnicek  ?

File: kbcstorage/tables.py [hunk 1]
```diff
@@ -42,11 +39,28 @@ def list(self, include=None):
...
+        Returns:
+            response_body: The parsed json from the HTTP response.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+
+        url = '{}/{}/tables'.format(self.base_url, bucket_id)
+        params = {'include': ','.join(include)}
+        return self._get(url, params=params).json()
```
Comment (Ogaday) [inline kbcstorage/tables.py]: This API call is already implemented here https://github.com/keboola/sapi-python-client/blob/25633efff914ecd82913e0af923e5e4dfafabf94/kbcstorage/buckets.py#L39

which I think is the correct place for it. Even though in the [docs](http://docs.keboola.apiary.io/#reference/tables/tables-in-bucket) it's in the `tables` section, the url indicates the buckets endpoint.

**Comment by Ogaday** [comment]
I agree with @odinuv - not a big fan of callying `.json()` on 95% of responses. I would prefer a `_get_json` function which calls `_get`. That said, it's an internal refactoring, so I'd be happy to merge this and make a separate pull for that change later.

File: kbcstorage/tables.py (hunk 1 above)
Comment (odinuv) [reply kbcstorage/tables.py]: +1

**Comment by pocin** [comment]
Given that #37 is merged, I'll 
- add the `_get_json()` method 
- fix conflicts 
so we can merge this as well

**Comment by Ogaday** [comment]
@pocin do you want a hand making those change? If you've already got some commits in the works then don't worry, but otherwise I'll have some free time tomorrow to work on it. Let me know.

**Comment by pocin** [comment]
@Ogaday thanks for pinging me, here's the `_get` refactoring.
I am confused about `POST` methods, though. Do you really want a `_post_json()` method to keep it consistent? I find that a little missleading.

**Comment by pocin** [comment]
Oh wow, what have I done at 09fc50b? I thought I'd add a new commit by resolving the conflict through the GitHub UI, I definately do not want to merge `master` into this PR! 👎 Can anyone delete the commit? Or maybe we can somehow rebase when merging this PR?

I am so sorry guys!

**Comment by pivnicek** [comment]
@pocin I don't think it's a problem.  The master branch isn't affected until we merge this PR into it.
If you want to revert 09fc50b on this branch, `git revert 09fc50b` should do the trick I think.

**Comment by odinuv** [comment]
@pocin Actually, `Merge branch 'master' into pocin-refactor-rebased` is exactly what you want to do to resolve merge conflicts, everything is as it should be

**Comment by pocin** [comment]
Oh okay, good to hear that! So what about the `_post()` "dilemma"? Do you want `_post_json()`?

**Comment by odinuv** [comment]
The question is what would they do? the API does not accept JSONs yet It's still forms. Even for configurations the JSON is sent as string in form field (which is sort of weird, but that's how it is)

**Comment by pocin** [comment]
nono, I mean't how to handle the situation, when `_post` request returns `JSON`.  I mean it I should implement

```
def _post_json(*args, **kwargs):
        return self._post(*args, **kwargs).json()
```


I.e:

-if `self._post()` returned `JSON` respnose directly, it wouldn't be consistent with the `_get()` and `_get_json()` pair of methods
-if `self._post()` returns `Response` object, we need to handle it in the child Endpoint (which is ok for me, but then why we made the `_get_json()` method in the first place?

Do you see where I am going?

**Comment by odinuv** [comment]
oh, right, as AFAIK the only request which does not return JSON is the get preview API call. so to me that seems unnecesary to me. I understand the naming and consistency frustration, that's why i suggested - `_get` (return json) and `_get_raw` (return Response)

**Comment by pocin** [comment]
Oh I see! That was my longing for consistency speaking in the last post. In this case it's not justified at all. More commits coming soon :-)
```
_post() will return json
_get() wil return json (use _get_raw()) internally
_get_raw() will return raw
```

**Comment by Ogaday** [comment]
@pocin I like what I see :+1:

**Pull request merged**